lib/types.d.ts

```diff
@@ -0,0 +1,1766 @@
+}
+
+export class BinaryExpression extends Node {
+  type: "BinaryExpression";
+  operator: "+" | "-" | "/" | "%" | "*" | "**" | "&" | "|" | ">>" | ">>>" | "<<" | "^" | "==" | "===" | "!=" | "!==" | "in" | "instanceof" | ">" | "<" | ">=" | "<=";
+  left: Expression;
+  right: Expression;
+}
+
+export class Directive extends Node {
+  type: "Directive";
+  value: DirectiveLiteral;
+}
+
+export class DirectiveLiteral extends Node {
+  type: "DirectiveLiteral";
+  value: string;
+}
+
+export class BlockStatement extends Node {
+  type: "BlockStatement";
+  body: Array<Statement>;
+  directives?: Array<Directive>;
+}
+
+export class BreakStatement extends Node {
+  type: "BreakStatement";
+  label?: Identifier;
+}
+
+export class CallExpression extends Node {
+  type: "CallExpression";
+  callee: Expression;
+  arguments: Array<Expression | SpreadElement | JSXNamespacedName>;
+  optional?: true | false;
+  typeParameters?: TypeParameterInstantiation | TSTypeParameterInstantiation;
+}
+
+export class CatchClause extends Node {
+  type: "CatchClause";
+  param?: Identifier;
+  body: BlockStatement;
+}
+
+export class ConditionalExpression extends Node {
+  type: "ConditionalExpression";
+  test: Expression;
+  consequent: Expression;
+  alternate: Expression;
+}
+
+export class ContinueStatement extends Node {
+  type: "ContinueStatement";
+  label?: Identifier;
+}
+
+export class DebuggerStatement extends Node {
+  type: "DebuggerStatement";
+}
+
+export class DoWhileStatement extends Node {
+  type: "DoWhileStatement";
+  test: Expression;
+  body: Statement;
+}
+
+export class EmptyStatement extends Node {
+  type: "EmptyStatement";
+}
+
+export class ExpressionStatement extends Node {
+  type: "ExpressionStatement";
+  expression: Expression;
+}
+
+export class File extends Node {
+  type: "File";
+  program: Program;
+  comments: any;
+  tokens: any;
+}
+
+export class ForInStatement extends Node {
+  type: "ForInStatement";
+  left: VariableDeclaration | LVal;
+  right: Expression;
+  body: Statement;
+}
+
+export class ForStatement extends Node {
+  type: "ForStatement";
+  init?: VariableDeclaration | Expression;
+  test?: Expression;
+  update?: Expression;
+  body: Statement;
+}
+
+export class FunctionDeclaration extends Node {
+  type: "FunctionDeclaration";
+  id?: Identifier;
+  params: Array<LVal>;
+  body: BlockStatement;
+  generator?: boolean;
+  async?: boolean;
+  declare?: boolean;
+  returnType?: TypeAnnotation | TSTypeAnnotation | Noop;
+  typeParameters?: TypeParameterDeclaration | TSTypeParameterDeclaration | Noop;
+}
+
+export class FunctionExpression extends Node {
+  type: "FunctionExpression";
+  id?: Identifier;
+  params: Array<LVal>;
+  body: BlockStatement;
+  generator?: boolean;
+  async?: boolean;
+  returnType?: TypeAnnotation | TSTypeAnnotation | Noop;
+  typeParameters?: TypeParameterDeclaration | TSTypeParameterDeclaration | Noop;
+}
+
+export class Identifier extends Node {
+  type: "Identifier";
+  name: any;
```

Comment:
https://github.com/babel/babel/blob/d25918aa5db682581a94536ff5b2abb615438365/packages/babel-types/src/definitions/core.js#L385-L390

Can we also assert that this is a string?
e.g.
```js
{
  validate: chain(
    assertValueType("string"),
    function (node, key, val) { 
      if (!isValidIdentifier(val)) { 
        // throw new TypeError(`"${val}" is not a valid identifer name`); 
      } 
    },
}
```

Reply:
Done